Test failure:  test_timeline_deletion_with_files_stuck_in_upload_queue

https://neon-github-public-dev.s3.amazonaws.com/reports/main/debug/3932228964/index.html#suites/837740b64a53e769572c4ed7b7a7eeeb/fbc2f2d387a430d5/

```
test_runner/regress/test_remote_storage.py:626: in test_timeline_deletion_with_files_stuck_in_upload_queue
    assert get_queued_count(file_kind="index", op_kind="upload") is None
E   AssertionError: assert 0 is None
E    +  where 0 = <function test_timeline_deletion_with_files_stuck_in_upload_queue.<locals>.get_queued_count at 0x7f4e11238af0>(file_kind='index', op_kind='upload')
```

<details><summary>Teardown error stacktrace captured from github runner logs</summary>

```
_ ERROR at teardown of test_timeline_deletion_with_files_stuck_in_upload_queue[local_fs] _
[gw0] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/_pytest/runner.py:311: in from_call
    result: Optional[TResult] = func()
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/_pytest/runner.py:255: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py:265: in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/pluggy/_manager.py:80: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/_pytest/threadexception.py:90: in pytest_runtest_teardown
    yield from thread_exception_runtest_hook()
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/_pytest/threadexception.py:75: in thread_exception_runtest_hook
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
E   pytest.PytestUnhandledThreadExceptionWarning: Exception in thread Thread-3
E   
E   Traceback (most recent call last):
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
E       httplib_response = self._make_request(
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 449, in _make_request
E       six.raise_from(e, None)
E     File "<string>", line 3, in raise_from
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 444, in _make_request
E       httplib_response = conn.getresponse()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/http/client.py", line 1347, in getresponse
E       response.begin()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/http/client.py", line 307, in begin
E       version, status, reason = self._read_status()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/http/client.py", line 276, in _read_status
E       raise RemoteDisconnected("Remote end closed connection without"
E   http.client.RemoteDisconnected: Remote end closed connection without response
E   
E   During handling of the above exception, another exception occurred:
E   
E   Traceback (most recent call last):
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/requests/adapters.py", line 489, in send
E       resp = conn.urlopen(
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 787, in urlopen
E       retries = retries.increment(
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/util/retry.py", line 550, in increment
E       raise six.reraise(type(error), error, _stacktrace)
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/packages/six.py", line 769, in reraise
E       raise value.with_traceback(tb)
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
E       httplib_response = self._make_request(
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 449, in _make_request
E       six.raise_from(e, None)
E     File "<string>", line 3, in raise_from
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 444, in _make_request
E       httplib_response = conn.getresponse()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/http/client.py", line 1347, in getresponse
E       response.begin()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/http/client.py", line 307, in begin
E       version, status, reason = self._read_status()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/http/client.py", line 276, in _read_status
E       raise RemoteDisconnected("Remote end closed connection without"
E   urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
E   
E   During handling of the above exception, another exception occurred:
E   
E   Traceback (most recent call last):
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/threading.py", line 954, in _bootstrap_inner
E       self.run()
E     File "/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/threading.py", line 892, in run
E       self._target(*self._args, **self._kwargs)
E     File "/__w/neon/neon/test_runner/regress/test_remote_storage.py", line 595, in checkpoint_thread_fn
E       client.timeline_checkpoint(tenant_id, timeline_id)
E     File "/__w/neon/neon/test_runner/fixtures/neon_fixtures.py", line 1345, in timeline_checkpoint
E       res = self.put(
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/requests/sessions.py", line 647, in put
E       return self.request("PUT", url, data=data, **kwargs)
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/requests/sessions.py", line 587, in request
E       resp = self.send(prep, **send_kwargs)
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/requests/sessions.py", line 701, in send
E       r = adapter.send(request, **kwargs)
E     File "/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/requests/adapters.py", line 547, in send
E       raise ConnectionError(err, request=request)
E   requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

https://github.com/neondatabase/neon/actions/runs/3932228964/jobs/6724673295#step:4:926

</details>

So perhaps the assertion failed (`0 != None`) and then the stacktrace within `<details>` is what failed after pageserver was shutdown fast. Assigning flaky as this didn't happen before the merge.

Different error in https://neon-github-public-dev.s3.amazonaws.com/reports/main/debug/4235679795/index.html#suites/837740b64a53e769572c4ed7b7a7eeeb/37ceeb4d54da4cd7/:

```
AssertionError: assert not ['2023-02-21T18:29:29.956571Z ERROR Error processing HTTP request: InternalServerError(timeline is Stopping\n']
```

Seems the error has now changed: https://neon-github-public-dev.s3.amazonaws.com/reports/release/debug/4243430314/index.html#suites/837740b64a53e769572c4ed7b7a7eeeb/b255552938221fb9/ again:

```
AssertionError: assert not ['2023-02-22T14:09:48.014410Z ERROR Error processing HTTP request: InternalServerError(timeline is Stopping\n']
```

--- a/test_runner/regress/test_remote_storage.py
+++ b/test_runner/regress/test_remote_storage.py
@@ -618,6 +618,9 @@ def test_timeline_deletion_with_files_stuck_in_upload_queue(
     # checkpoint operations. Hence, checkpoint is allowed to fail now.
     log.info("sending delete request")
     checkpoint_allowed_to_fail.set()
+    env.pageserver.allowed_errors.append(
+        ".+ERROR Error processing HTTP request: InternalServerError\\(timeline is Stopping.+"
+    )
     client.timeline_delete(tenant_id, timeline_id)
 
     assert not timeline_path.exists()

--- a/test_runner/regress/test_remote_storage.py
+++ b/test_runner/regress/test_remote_storage.py
@@ -619,7 +619,7 @@ def test_timeline_deletion_with_files_stuck_in_upload_queue(
     log.info("sending delete request")
     checkpoint_allowed_to_fail.set()
     env.pageserver.allowed_errors.append(
-        ".+ERROR Error processing HTTP request: InternalServerError\\(timeline is Stopping.+"
+        ".+ERROR Error processing HTTP request: InternalServerError\\(timeline is Stopping"
     )
     client.timeline_delete(tenant_id, timeline_id)
 

